velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -125,6 +126,18 @@ std::shared_ptr<::parquet::WriterProperties> getArrowParquetWriterOptions(
   return properties->build();
 }
 
+namespace {
+
+void exportToArrow(
+    const TypePtr& type,
+    std::shared_ptr<memory::MemoryPool> pool,
+    ArrowSchema& out) {
+  auto leafPool = pool->addLeafChild("parquet-write-schema-convert");
+  exportToArrow(BaseVector::create(type, 0, leafPool.get()), out);
```

Comment:
There was a question by Masha if we need to create a vector here. Looking at  `void exportToArrow(const VectorPtr& vec, ArrowSchema& arrowSchema)` inside Bridge.cpp, I think adding a new API `exportToArrow(const TypePtr& type, ArrowSchema& arrowSchema)` will result in code duplication.

Reply:
@mbasmanova  @majetideepak  Can we adding new API `exportToArrow(const TypePtr& type, ArrowSchema& arrowSchema)` in the following PR. Merge this PR firstly.

Reply:
My preference would be to first add the new API (in a separate PR).

Reply:
@majetideepak @mbasmanova Filed https://github.com/facebookincubator/velox/pull/6627 to add exportToArrow(const TypePtr& type, ArrowSchema& arrowSchema) API in Bridge.

Reply:
As per the discussion in #6627, it seems to me that we cannot create an empty Vector wrapper to define an arrowSchema here. The recordBatch might have encodings.
Instead, we should pass the type as an alias to `exportToArrow(const VectorPtr& vec, ArrowSchema& arrowSchema, const TypePtr& alias = nullptr)` and rename the fields.
CC: @Yuhta

Reply:
Can we rename the fields in vector that is passed to arrow bridge?

Reply:
@majetideepak @Yuhta I'm having trouble understanding why renaming a field is necessary, considering that the type has already been provided when creating the vector.

Reply:
@JkSelf, the proposed solution for the custom schema below will not work. Looking at the bridge.cc in Arrow, the schema inference depends on the Vector encodings as well. For example, a dictionary vector returns a DictionaryType which is not in Velox.
As @Yuhta suggested, the right way is to rename the `data` fields in Writer::write() with the Velox Type provided and keep the current Arrow API invocation.
```
 if (options.schema) {
    ArrowSchema arrowSchema;
    exportToArrow(options.schema, pool_, arrowSchema);

    PARQUET_ASSIGN_OR_THROW(
        arrowContext_->schema, ::arrow::ImportSchema(&arrowSchema));
......
```

Reply:
Unfortunately, there are no parquet writer tests to catch such issues. We started adding some tests this week.